Servers/Administration/OJS.Servers.Administration/Controllers/SubmissionsController.cs

```diff
@@ -226,7 +228,28 @@ protected override async Task DeleteEntityAndSaveAsync(Submission submission, Ad
             return x => x.Id == submissionId;
         }
 
-        return base.MasterGridFilter;
+        ParameterExpression parameter = Expression.Parameter(typeof(Submission), "x");
+
+        Expression<Func<Submission, bool>> firstExpression = x =>
+        (
+            x.Problem.ProblemGroup.Contest.LecturersInContests.Any(l => l.LecturerId == this.User.GetId()) ||
+            x.Problem.ProblemGroup.Contest.Category!.LecturersInContestCategories.Any(lc => lc.LecturerId == this.User.GetId()) ||
+            this.User.IsAdmin());
```

Comment:
Better to have the `this.User.IsAdmin()` check first, so no unnecessary joins will be made if the user is admin..

Reply:
This is reworked in a business service